feat: Add CI/CD Workflow

sudiptog81 opened the pull request:
Added CI/CD Workflow with GitHub Actions. There are certain libraries and dependencies that need to be built/installed in order to build Tracee from source easily, I have included those steps as well. I have used the existing routines in the `Makefile` for the various jobs.

* Workflows `build` and `test` shall be triggered on any push to the repository on any branch. As currently there is a single branch, this is fine as per my conscience.
* Workflow `release` shall be triggered only when a new tag is pushed to the repository. 

The [`build`](https://github.com/sudiptog81/tracee/runs/1251932885) workflow ran fine on my branch, however, the [`test`](https://github.com/sudiptog81/tracee/runs/1251932887) workflow failed, owing to #252. Please advise on that.

The workflows may be rudimentary and may need changes but can be easily extended in the future. Fixes #254.

itaysk commented on .github/workflows/test.yaml:
```diff
@@ -0,0 +1,45 @@
+name: test
```
I think it would be better to merge this WF with the "build" WF, the build step takes alot of time and rebuilding here again for the test seems redundant to me. what I'm suggesting is to just take the `make test` step from here and move it into the `build` workflow (which would be renamed to build-test)

sudiptog81 replied:
Sure! Addressing this in the next commit.

itaysk commented on .github/workflows/release.yaml:
```diff
@@ -0,0 +1,39 @@
+name: release
```
This will not work right now since goreleaser will miss the credentials to push to dockerhub. If you want to merge this PR quickly, we can remove this publish WF from this PR and we merge it without it. We can add publish in a new PR

sudiptog81 replied:
I will remove the release workflow from this PR for now.

itaysk commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,45 @@
+          make
+          sudo make install
+
+      - name: Lint
```
Lint should come before the Install Libraries step, no reason to wait for the prerequisites to install before linting

sudiptog81 replied:
Sure! Addressing this in the next commit.

itaysk commented on .github/workflows/build.yaml:
```diff
@@ -0,0 +1,45 @@
+      - name: Checkout Code
+        uses: actions/checkout@v2
+
+      - name: Install Libraries
```
We can simplify this job and eliminate this Install Libraries step if we just build using the docker containers. so, instead of using `make build` we would use `make build-docker`

sudiptog81 replied:
Running tests requires BCC (has to be built from source), so I believe, we have to install the dependencies.

itaysk replied:
My reservation with this approach is 
1. building bcc from source every time is inconsistent, when we release we use a specific commit of bcc that we know works for us.
2. It takes a while to install all these packages and if we use a docker image its all contained in the image
3. Potential code duplication between this script and the Dockerfile.

This is why I made the suggestion. You are correct that to test we will need bcc, but we can also test in a container. For now the tests are broken anyway, and also there's WIP to remove bcc from Tracee, so we can defer the decision for later. 
For now I think it would be best to do a build-docker in this WF and later we can improve it if needed (I also want to merge this by the end of the month to make you eligible for hacktoberfest)

sudiptog81 replied:
Got your point! Will make the required changes.

itaysk reviewed (changes requested):
@sudiptog81 thanks for working on this, and sorry I was delayed with the review. Added a few suggestions for improvements

sudiptog81 commented:
@itaysk I addressed most of the comments, however, I found running the tests requires BCC which has to be built from source, and therefore, I believe we should keep that step. Let me know your opinion.

itaysk reviewed (approved):
Thank you @sudiptog81 ,
sorry we had to scope down your original PR. we can continue with the tests and with the release in separate PRs
